Libraries/Settings/Settings.js

```diff
@@ -11,25 +11,44 @@
  */
 'use strict';
 
-var RCTDeviceEventEmitter = require('RCTDeviceEventEmitter');
-var RCTSettingsManager = require('NativeModules').SettingsManager;
+const RCTDeviceEventEmitter = require('RCTDeviceEventEmitter');
+const RCTSettingsManager = require('NativeModules').SettingsManager;
 
-var invariant = require('fbjs/lib/invariant');
+const invariant = require('fbjs/lib/invariant');
 
-var subscriptions: Array<{keys: Array<string>, callback: ?Function}> = [];
+const subscriptions: Array<{keys: Array<string>; callback: ?Function}> = [];
 
-var Settings = {
+/**
+ * @class
+ * @description
+ * An interface to the native "preferences" file storage that uses simple key-value pairs
+ * 
+ */
+const Settings = {
   _settings: RCTSettingsManager && RCTSettingsManager.settings,
 
   get(key: string): mixed {
     return this._settings[key];
   },
 
-  set(settings: Object) {
+ /**
+   * Sets the value for a name-value pairs
+   * @param settings the object with the name/value pairs to set
+   *
+   * Note that on android the only allowed value types are number, string and boolean
+   */
+  set(settings: Object): void {
     this._settings = Object.assign(this._settings, settings);
```

Comment:
I didn't actually change any code in `Settings.js` (other than rename the file from `Settings.ios.js`). But I agree we should probably improve this file's code as part of this change.